### PR TITLE
Remove last rst docstrings

### PR DESCRIPTION
**Pull request by hayd**
The second commit moves collections docs inline.

cc @MichaelHatherly #14378

Hopefully I have the ref syntax correct.

### CODE REVIEW

**Comment by MichaelHatherly** [comment]
Thanks @hayd. Ref syntax looks correct. Have you also done a `genstdlib.jl` run to check nothing got missed when moving them?

File: base/collections.jl [hunk 1]
```diff
@@ -82,6 +90,11 @@ end
 
 
 # Turn an arbitrary array into a binary min-heap in linear time.
+"""
+    heapify!(v, [ord])
+
+In-place [``heapify``](:func:`heapify`).
```
Comment (MichaelHatherly) [inline base/collections.jl]: Should be single backticks for code. Double backticks now produce latex. Also in the `PriorityQueue` and `sprand` below.

Comment (hayd) [inline base/collections.jl]: that explains the genstdlib differences! will fix

**Comment by hayd** [comment]
when doing genstdlib I get:

```
ErrorException("Cannot find doc for Base.Collections.PriorityQueue{K,V,O<:Base.Order.Ordering}")
WARNING: Mod Base.Collections PriorityQueue
INFO: stdlib/collections.rst: no docs for PriorityQueue(K, V, [ord]) in Base.Collections
```
any idea why that might happen? repl help works.

**Comment by MichaelHatherly** [comment]
I'm not getting that error locally, does your julia include https://github.com/JuliaLang/julia/pull/14513? That should fix that problem I believe. (`genstdlib.jl` wasn't looking in the `.main` field of `TypeDoc`s for documentation.)

**Comment by hayd** [comment]
Ah, no it did not.

One thing that running the genstdlib seems to do is make the rst one-line, is that known/ok?

**Comment by MichaelHatherly** [comment]
> One thing that running the genstdlib seems to do is make the rst one-line, is that known/ok?

Yeah, that's expected.

**Comment by tkelman** [comment]
lgtm

**Comment by hayd** [comment]
I ran genstdlib to fix the PriorityQueue message above and got a appveyor fail. I can't see what failed.

ps. I have a PR coming to inline/remove a few other helpdb files.

**Comment by tkelman** [comment]
One of the workers hung and timed out. Almost certainly unrelated, but restarted just to be sure.

> ps. I have a PR coming to inline/remove a few other helpdb files.

cool

**Pull request merged**